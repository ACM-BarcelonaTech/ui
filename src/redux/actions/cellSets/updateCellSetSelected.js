import { CELL_SETS_SET_SELECTED } from '../../actionTypes/cellSets';

const updateCellSetSelected = (keys, tab) => async (dispatch, getState) => {
  const {
    loading, error,
  } = getState().cellSets;

  if (loading || error) {
    return null;
  }

  await dispatch({
    type: CELL_SETS_SET_SELECTED,
    payload: {
      keys,
      tab,
    },
  });
};

export default updateCellSetSelected;
