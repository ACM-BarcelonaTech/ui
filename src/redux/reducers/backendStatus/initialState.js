const initialExperimentBackendStatus = {
  loading: false,
  error: false,
  status: {
  },
};

const initialState = {};

export default initialState;

export { initialExperimentBackendStatus };
