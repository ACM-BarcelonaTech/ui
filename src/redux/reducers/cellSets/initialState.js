const initialState = {
  properties: {},
  hierarchy: [],
  selected: [],
  loading: true,
  error: false,
};

export default initialState;
