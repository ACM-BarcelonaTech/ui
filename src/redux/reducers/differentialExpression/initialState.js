const initialState = {
  properties: {
    data: [],
    comparisonGroup: {},
    comparisonType: null,
    loading: false,
    error: false,
    total: 0,
  },
  comparison: {
    type: 'within',
    advancedFilters: [],
    group: {
      between: {
        cellSet: null,
        compareWith: null,
        basis: null,
      },
      within: {
        cellSet: null,
        compareWith: null,
        basis: null,
      },
    },
    ordering: {
      orderBy: 'logFC',
      orderDirection: 'DESC',
    },
  },
};

export default initialState;
