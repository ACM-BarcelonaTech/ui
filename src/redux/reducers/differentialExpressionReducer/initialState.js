const initialState = {
  properties: {
    data: {},
    cellSets: {},
    loading: false,
    error: false,
  },
};

export default initialState;
