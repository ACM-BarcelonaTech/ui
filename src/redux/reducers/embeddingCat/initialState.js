const initialState = {
  width: 700,
  height: 550,
  pointSize: 5,
  toggleInvert: '#FFFFFF',
  masterColour: '#000000',
  umap1Domain: null,
  umap2Domain: null,
  titleText: '',
  titleSize: 20,
  titleAnchor: 'start',
  axisTitlesize: 13,
  axisTicks: 13,
  transGrid: 0,
  axesOffset: 10,
  masterFont: 'sans-serif',
  xaxisText: 'UMAP 1',
  yaxisText: 'UMAP 2',
  pointStyle: 'circle',
  pointOpa: 5,
  g1Color: 'red',
  g2mColor: 'green',
  sColor: 'blue',
  legendTextColor: '#FFFFFF',
  legendEnabled: true,
  legend: null,
  geneexpLegendloc: '',
  labelSize: 28,
  labelShow: 1,
  labelFont: 2,
  labelsEnabled: true,
  selectedClusters: [],
  testVar: null,
};

export default initialState;
