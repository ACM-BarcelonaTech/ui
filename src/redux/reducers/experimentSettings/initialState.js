const metaInitialState = {
  loading: false,
  completingStepError: false,
  loadingSettingsError: false,
  selectedConfigureEmbeddingPlot: 'cellCluster',
  changedQCFilters: new Set(),
};

const initialState = {
  info: {
    experimentId: null,
    experimentName: null,
    sampleIds: [],
    paramsHash: '',
  },
  processing: {
    meta: metaInitialState,
  },
  originalProcessing: {},
};

export { metaInitialState };
export default initialState;
