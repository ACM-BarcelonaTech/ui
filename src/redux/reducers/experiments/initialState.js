const experimentTemplate = {
  projectUuid: null,
  name: null,
  description: 'Analysis description',
  id: null,
  createdDate: null,
  lastViewed: null,
  notifyByEmail: false,
  meta: { organism: null, type: '10x' },
  sampleIds: [],
};

const initialState = {
  meta: {
    loading: false,
    error: false,
    saving: false,
  },
  ids: [],
};

export default initialState;
export { experimentTemplate };
