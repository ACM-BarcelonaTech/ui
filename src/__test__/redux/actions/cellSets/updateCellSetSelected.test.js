import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import fetchMock, { enableFetchMocks } from 'jest-fetch-mock';
import updateCellSetSelected from '../../../../redux/actions/cellSets/updateCellSetSelected';
import initialState from '../../../../redux/reducers/cellSetsReducer/initialState';

enableFetchMocks();
const mockStore = configureStore([thunk]);

describe('updateCellSetSelected action', () => {
  const experimentId = '1234';
  const keys = [1, 3, 4, 5];

  beforeEach(() => {
    const response = new Response(JSON.stringify({}));

    fetchMock.resetMocks();
    fetchMock.doMock();
    fetchMock.mockResolvedValueOnce(response);
  });

  it('Does not dispatch on loading state', async () => {
    const store = mockStore({ cellSets: { loading: true, error: false } });
    store.dispatch(updateCellSetSelected(experimentId, keys));
    expect(store.getActions().length).toEqual(0);
  });

  it('Does not dispatch on error state', async () => {
    const store = mockStore({ cellSets: { loading: false, error: true } });
    store.dispatch(updateCellSetSelected(experimentId, keys));
    expect(store.getActions().length).toEqual(0);
  });

  it('Dispatches an action to update property to the reducer', async () => {
    const store = mockStore({ cellSets: { ...initialState, loading: false } });
    store.dispatch(updateCellSetSelected(experimentId, keys));

    const firstAction = store.getActions()[0];
    expect(firstAction).toMatchSnapshot();
  });
});
