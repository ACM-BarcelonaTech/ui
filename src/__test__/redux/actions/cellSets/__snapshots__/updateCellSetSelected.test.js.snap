// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updateCellSetSelected action Dispatches an action to update property to the reducer 1`] = `
Object {
  "payload": Object {
    "loadingColors": true,
  },
  "type": "cellSets/setSelected",
}
`;
